src/go/k8s/api/redpanda/v1alpha2/common.go

```diff
@@ -50,6 +51,12 @@ const (
 	SASLMechanismAWSManagedStreamingIAM SASLMechanism = config.SASLMechanismAWSManagedStreamingIAM
 )
 
+// Equals determines if the given SASLMechanism is equal to this
+// mechanism, it does so in a case-insensitive way.
+func (s SASLMechanism) Equals(other SASLMechanism) bool {
```

Comment:
I will forget to use this 100% of the time. This would be a good example case for custom linters via semgrep or ruleguard.